core/amber/src/main/protobuf/edu/uci/ics/amber/engine/architecture/rpc/controlcommands.proto

```diff
@@ -244,7 +244,8 @@ message AssignPortRequest {
   core.PortIdentity portId = 1 [(scalapb.field).no_box = true];
   bool input = 2;
   map<string, string> schema = 3;
-  string storageUri = 4;
+  repeated string storageUris = 4;
+  repeated sendsemantics.Partitioning partitionings = 5;
```

Comment:
here we are still counting on index to match uri and partitioning. better to change this design if it is easy. if it is hard, then nvm.

Reply:
I tried to implement a cleaner design but turned out It was not straightforward. I left a TODO for this in `RegionExecutionCoordinator`.